### PR TITLE
Slightly Clarify Race Condition Docs

### PR DESCRIPTION
**Pull request by toksfifo**
# Pull Request/Issue Resolution

#### Description of Change:
Adding a slight clarification to the documentation around race conditions.

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [x] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [x] Check that all links work.
- [x] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [x] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [x] Tag others as reviewers as necessary.
- [x] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @timothy-kim or @KellieHawks for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@Timothy-Kim</td>
    <td>Data Infrastructure<br>Platform Infrastructure<br>Quality Infrastructure</td>
  </tr>
  <tr>
    <td>@kelliehawks</td>
    <td>Internal Tools<br>Product Partnerships<br>SDKs<br>SMS</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Channels<br>FIX<br>Intelligence<br>Reporting<br>SMB</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Email (Composition and Infrastructure)<br>Ingestion<br>Messaging</td>
  </tr>
</table>
</details>

### CODE REVIEW

File: _docs/_user_guide/engagement_tools/testing/race_conditions.md [hunk 1]
```diff
@@ -25,15 +25,15 @@ However, in some cases, the second event will trigger first. This means that a m
...
+If you're using separate API endpoints to create users and trigger Canvases/campaigns, this can also result in this race condition. When user information is sent to Braze via the `users/track` endpoint, it may occasionally take a few seconds to process. As a result, when requests are made to the `users/track` and [messaging endpoints][4] at the same time, there is currently no guarantee that the user information will be updated before a message is sent. If these requests are made in the same API request, there should be no issue. Note that if you are sending a scheduled message API request, these requests must be separate, and a user must be created before sending the schdeuled API request.
```
Comment (parsonsj) [inline _docs/_user_guide/engagement_tools/testing/race_conditions.md:28]: I'm confused about "these requests"—requests to the messaging endpoints and `users/track` in the same API request? Maybe reword a little more?

File: _docs/_user_guide/engagement_tools/testing/race_conditions.md [hunk 2]
```diff
@@ -25,15 +25,15 @@ However, in some cases, the second event will trigger first. This means that a m
...
 
 {% alert note %}
-If user attributes are sent via SDK or in the same user/track call as the event, then Braze will automatically process those first before attempting to send any message.
+If user attributes are sent in the same request as the event (either from `users/track` or from the SDK), then Braze will generally process those first before attempting to send any message.
```
Comment (AshRagheb) [inline _docs/_user_guide/engagement_tools/testing/race_conditions.md:31]:
```suggestion
If user attributes and events are sent in the same request (either from `users/track` or from the SDK), then Braze will generally process attributes before events or attempting to send any message.
```

**Review by KellieHawks — APPROVED** [review]
